meson.build

```diff
@@ -0,0 +1,64 @@
+project('snitch', 'cpp',
+  default_options : ['cpp_std=c++20', 'default_library=static'],
+  version : '1.0.0'
+)
+
+subdir('snitch')
+
+headers = files(
+  'include/snitch/snitch.hpp',
+  'include/snitch/snitch_teamcity.hpp'
+)
+
+install_headers(headers, subdir: 'snitch')
```

Comment:
You might want to use `install_subdir` here instead.

Reply:
Thanks for the suggestion.

I tried `install_subdir` of `include/snitch`, then decided to stick with `install_headers`.
`install_subdir` has to exclude the `snitch_config.hpp.config` template file in that directory.
`install_headers` is more explicit in its intent, and the `headers` files var is already used a few times.

Then, in hacking this, I noticed that the single-header custom_target wasn't being installed. Oops.

To get the `custom_target` to work I had to move it to the `meson.build` in the new `./snitch` source subdir, already created in this PR to deal with the `configure_file` command.

In the cmake build `./snitch` is purely a build subdir, not present in the source tree.

In meson build the only way I found to get paths to work out right is to create a `./snitch` subdir in the source tree to reflect the desired path in the build tree. I'm likely missing something here; I don't see this restriction mentioned in the meson documentation. Is this a thing with `custom_target`, `configure_file` (and the new `preprocessor` target)?